RFC-0006-nan-definition.md

```diff
@@ -0,0 +1,179 @@
+
+We will also add thorough documentation for the new behavior.
+ 
+## Other Approaches Considered
+
+### IEEE-754 and throw an error for order by
+
+In this approach we follow the strict IEEE-754 definition for equality and comparisons. Meaning NaN = NaN would be false.  NaN less than or greater than anything is false and NaN <> NaN is true.  Any function on NaN input including min/max will return NaN. Grouping or distinct on NaN keys will return separate groups for each nan value. In cases where we could not simply return NaN, we would throw an exception.  That would mean that for order by on a double column containing NaN we would throw an error (as presumably would array_sort). 
+
+This has the advantage of adhering to IEEE. It also provides a consistent definition of our operators to reduce chances of bugs (no difference between results you would get from applying < or > vs. what you would get from doing a sort).
+
+A disadvantage is that users are very unlikely to want to get an error for order by just because their data has NaN in it- especially if we otherwise support NaN values. it also still leaves the funny behavior with maps/hashing where you can put things in a map but can't get it out. There are also no other systems that behave this way.
+
+### IEEE-754 with total order
+
+In this option we implement the full IEEE spec including the total order definition to use for sorting and order by.  NaN = NaN would be false.  NaN less than or greater than anything is false and NaN <> NaN is true. Group bys and distincts would return separate groups for each NaN value.  Order by on a double column containing NaN would give you the IEEE defined total order.  An open question would be the behavior of functions like min or max (are they functions on NaN input that should return NaN or are they sorting functions that should return based on the total order? what about when it's a topn function like min(x, n) or max(x, n)?).  
+
+Advantages here are full compliance with the IEEE spec.
+
+Disadvantages are the inconsistency between comparison and sorting operators can lead to confusion and bugs. It keeps the funny behavior with maps where you can put nans in but can't get them out (and not sure if it would cause implementation problems if we did group bys this way). It does not seem like other dbs do this either.
```

Comment:
```suggestion
Disadvantages are the inconsistency between comparison and sorting operators can lead to confusion and bugs. It keeps the funny behavior with maps where you can put NaNs in but can't get them out (and not sure if it would cause implementation problems if we did group bys this way). It does not seem like other dbs do this either.
```
Also, suggest expanding "dbs".